lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/util/AutoCloseableCloser.java

```diff
@@ -26,6 +30,9 @@
 public final class AutoCloseableCloser
         implements AutoCloseable
 {
+    @GuardedBy("this")
+    private boolean closed;
+    @GuardedBy("this")
     private final Deque<AutoCloseable> stack = new ArrayDeque<>(4);
```

Comment:
I think this can be an ArrayList now... if you reverse the list in the close method

Reply:
Done